src/inference/src/compilation_context.cpp

```diff
@@ -46,10 +46,12 @@ static int32_t as_int32_t(T v) {
 std::string NetworkCompilationContext::calculateFileInfo(const std::string& filePath) {
     uint64_t seed = 0;
     auto absPath = filePath;
-    try {
-        absPath = FileUtils::absoluteFilePath(filePath);
-    } catch (...) {
-        // can't get absolute path, will use filePath for hash
+    if (filePath.size() > 0) {
```

Comment:
why do we even call `calculateFileInfo` method for cases with no file?

Reply:
All model cache blob have a header that is composed of IEversion and FileInfo. It is generated as below in `compile_model_impl()` regardless of existence of `modelPath`.
https://github.com/openvinotoolkit/openvino/blob/b4faeebc84e65dc9e418caf8bb44e143bfcf8246/src/inference/src/ie_core.cpp#L427-L431
I'm not the original author of this code, so I'm not sure what the original intent was. However, I think recording the same value for a null `modelPath` has the same effect as recording nothing. And I chose to record the same value for a null `modelPath` in order to make code changes as minimal as possible.

Reply:
thanks for the explanation.
Could you please also port the fix for 2022.3?